front-end/src/components/KeyMeasureDiagnostic/index.vue

```diff
@@ -31,19 +31,27 @@
       QualityMeasureDiagnostic,
       WasteMeasureDiagnostic
     },
-    props: ['measure', 'closeModal', 'originalDiagnostic'],
+    props: ['measure', 'closeModal', 'originalDiagnostics'],
     data() {
       return {
-        diagnostic: this.originalDiagnostic,
         measureDiagnosticComponentName: this.measure.baseComponent + "Diagnostic",
       };
     },
     methods: {
-      submit() {
-        saveDiagnostic(this.measure.id, this.diagnostic);
+      async submit() {
+        (this.measureDiagnosticComponentName === 'QualityMeasureDiagnostic') ?
+          await saveDiagnostics(this.diagnostics) :
+          await saveDiagnostic(this.diagnostics);
```

Comment:
Avec les autres changements proposés je pense qu'on pourrait faire directement `await saveDiagnostics(this.diagnostics)` ici